httpx/client.py

```diff
@@ -169,19 +170,24 @@ def merge_cookies(
             # We perform these checks here, so that calls to `response.next()`
             # will raise redirect errors if appropriate.
             if len(history) > self.max_redirects:
-                raise TooManyRedirects()
+                raise TooManyRedirects(response=history[-1])
             if request.url in [response.url for response in history]:
-                raise RedirectLoop()
+                raise RedirectLoop(response=history[-1])
+
+            try:
+                response = await self.dispatch.send(
+                    request, verify=verify, cert=cert, timeout=timeout
+                )
+            except HttpxError as exc:
```

Comment:
I think it might be better to include this in the `AsyncClient.send()` method instead, wrapping the `response = await self.send_handling_redirects(...)` call.